kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorIT.java

```diff
@@ -0,0 +1,524 @@
+       * DNS-based bucket access scheme
+       * resulting in attempts to connect to addresses like "bucketname.localhost"
+       * which requires specific DNS setup.
+       */
+      EndpointConfiguration endpoint = new EndpointConfiguration(MOCK_S3_URL, AWS_REGION);
+      AmazonS3 mockClient = AmazonS3ClientBuilder
+          .standard()
+          .withPathStyleAccessEnabled(true)
+          .withEndpointConfiguration(endpoint)
+          .withCredentials(new AWSStaticCredentialsProvider(new AnonymousAWSCredentials()))
+          .build();
+
+      log.info("No credentials found, using mock S3 client.");
+      return mockClient;
+    } else {
+      log.info("Credentials found, using real S3 client.");
+      // DefaultAWSCredentialsProviderChain,
+      // assumes .aws/credentials is setup and test bucket exists
+      return AmazonS3ClientBuilder.standard().withRegion(AWS_REGION).build();
+    }
+  }
+
+  /**
+   * Clear the given S3 bucket. Removes the contents, keeps the bucket.
+   *
+   * @param bucketName the name of the bucket to clear.
+   */
+  private static void clearBucket(String bucketName) {
+    for (S3ObjectSummary file : S3Client.listObjectsV2(bucketName).getObjectSummaries()) {
+      S3Client.deleteObject(bucketName, file.getKey());
+    }
+  }
+
+  /**
+   * Check if the file names in the bucket have the expected namings.
+   *
+   * @param bucketName        the name of the bucket with the files
+   * @param expectedFiles     the list of expected filenames for exact comparison
+   * @return whether all the files in the bucket match the expected values
+   */
+  private boolean fileNamesValid(String bucketName, List<String> expectedFiles) {
+    List<String> actualFiles = new ArrayList<>();
+    for (S3ObjectSummary file : S3Client.listObjectsV2(bucketName).getObjectSummaries()) {
+      actualFiles.add(file.getKey());
+    }
+    return expectedFiles.equals(actualFiles);
+  }
+
+  /**
+   * Check the contents of the files in the S3 bucket compared to the expected row.
+   *
+   * @param bucketName          the name of the s3 test bucket
+   * @param expectedRowsPerFile the number of rows a file should have
+   * @param expectedRow         the expected row data in each file
+   * @return whether every row of the files read equals the expected row
+   */
+  private boolean fileContentsAsExpected(String bucketName, int expectedRowsPerFile,
+      Struct expectedRow) {
+    log.info("expectedRow: {}", expectedRow);
+    for (S3ObjectSummary file : S3Client.listObjectsV2(bucketName).getObjectSummaries()) {
+      String destinationPath = TEST_DOWNLOAD_PATH + file.getKey();
+      File downloadedFile = new File(destinationPath);
+      log.info("Saving file to : {}", destinationPath);
+      S3Client.getObject(new GetObjectRequest(bucketName, file.getKey()), downloadedFile);
+
+      String fileExtension = new S3FileInfo(file.getKey()).extension;
```

Comment:
should just extract the extension here so we can get rid of this class since its only being used for the extension now